// smss/es/colors/index.js
// @flow
import common from './common';
import red from './red';
import pink from './pink';
import purple from './purple';
import deepPurple from './deepPurple';
import indigo from './indigo';
import blue from './blue';
import lightBlue from './lightBlue';
import cyan from './cyan';
import teal from './teal';
import green from './green';
import lightGreen from './lightGreen';
import lime from './lime';
import yellow from './yellow';
import amber from './amber';
import orange from './orange';
import deepOrange from './deepOrange';
import brown from './brown';
import grey from './grey';
import blueGrey from './blueGrey';
export { common, red, pink, purple, deepPurple, indigo, blue, lightBlue, cyan, teal, green, lightGreen, lime, yellow, amber, orange, deepOrange, brown, grey, blueGrey, };
export default {
    common: common,
    red: red,
    pink: pink,
    purple: purple,
    deepPurple: deepPurple,
    indigo: indigo,
    blue: blue,
    lightBlue: lightBlue,
    cyan: cyan,
    teal: teal,
    green: green,
    lightGreen: lightGreen,
    lime: lime,
    yellow: yellow,
    amber: amber,
    orange: orange,
    deepOrange: deepOrange,
    brown: brown,
    grey: grey,
    blueGrey: blueGrey,
};
