// smss/es/utils/index.js
export function noop() {
} // tslint:disable-line
export function same(o) {
    return o;
}
export function is(x, y) {
    if (x === y) {
        return x !== 0 || 1 / x === 1 / y;
    }
    else {
        return x !== x && y !== y;
    }
}
export function range(size, start, step) {
    if (size === void 0) { size = 50; }
    if (start === void 0) { start = 0; }
    if (step === void 0) { step = 1; }
    var arr = [];
    for (var i = 0; i < size; i += 1) {
        arr[i] = i + start;
    }
    return arr;
}
export function oneOf(actual, expected) {
    if (!Array.isArray(expected)) {
        return new Error('Invalid argument supplied to oneOfType: expected an instance of array');
    }
    if (expected.filter(function (value) { return is(actual, value); }).length === 0) {
        return false;
    }
    return true;
}
export function o2a(obj) {
    return Object.keys(obj).map(function (key) { return ({ key: key, value: obj[key] }); });
}
export function a2o(arr) {
    return arr.length === 0 || arr.length === 1
        ? (_a = {}, _a[arr[0].key] = arr[0].value, _a) : arr.reduce(function (a, b) {
        return Object.assign(a, (_a = {}, _a[b.key] = b.value, _a)); // tslint:disable-line
        var _a;
    }, {});
    var _a;
}
export function isPromise(value) {
    if (value !== null && typeof value === 'object') {
        return value && typeof value.then === 'function';
    }
    return false;
}
export function debounce() {
    // todo
}
export function throttle() {
    // todo
}
