// smss/es/style/index.js
// @大小
export var s = function (flex) { return ({ flex: flex }); };
export var w = function (width) { return ({ width: width }); };
export var h = function (height) { return ({ height: height }); };
export var wh = function (width, height) { return ({ width: width, height: height || width }); };
// @颜色
export var c = function (color) { return ({ color: color }); };
// @背景
export var b = function (backgroundColor) { return ({ backgroundColor: backgroundColor }); };
// @字体
export var f = function (fontSize) { return ({ fontSize: fontSize }); };
export var lh = function (lineHeight) { return ({ lineHeight: lineHeight }); };
// @内边距
export var pd = function (padding) { return ({ padding: padding }); };
export var pt = function (paddingTop) { return ({ paddingTop: paddingTop }); };
export var pr = function (paddingRight) { return ({ paddingRight: paddingRight }); };
export var pb = function (paddingBottom) { return ({ paddingBottom: paddingBottom }); };
export var pl = function (paddingLeft) { return ({ paddingLeft: paddingLeft }); };
export var plr = function (paddingLeft, paddingRight) {
    return ({ paddingLeft: paddingLeft, paddingRight: paddingRight || paddingLeft });
};
export var ptb = function (paddingTop, paddingBottom) {
    return ({ paddingTop: paddingTop, paddingBottom: paddingBottom || paddingTop, });
};
// @外边距
export var mg = function (margin) { return ({ margin: margin }); };
export var mt = function (marginTop) { return ({ marginTop: marginTop }); };
export var mr = function (marginRight) { return ({ marginRight: marginRight }); };
export var mb = function (marginBottom) { return ({ marginBottom: marginBottom }); };
export var ml = function (marginLeft) { return ({ marginLeft: marginLeft }); };
export var mlr = function (marginLeft, marginRight) { return ({ marginLeft: marginLeft, marginRight: marginRight || marginLeft }); };
export var mtb = function (marginTop, marginBottom) { return ({ marginTop: marginTop, marginBottom: marginBottom || marginTop }); };
// @位置
export var ps = function (top, right, bottom, left) { return ({ top: top, right: right, bottom: bottom, left: left }); };
