// smss/webpack.config.js
const path = require('path')
const webpack = require('webpack')
const HtmlWebpackPlugin = require('html-webpack-plugin')

const env = process.env.NODE_ENV || 'development'
const port = 3000
const publicPath = '/'

const src = path.resolve(__dirname, './src')
const lib = path.resolve(__dirname, './node_modules')
const dst = path.resolve(__dirname, './dist')

const development = {
  devtool: 'eval-cheap-module-source-map',
  devServer: {
    port,
    host: '0.0.0.0',
    https: false,
    compress: true,
    disableHostCheck: true,
    contentBase: dst,
  },
  context: `${src}`,
  resolve: {
    extensions: ['.js', '.ts', '.tsx'],
    // modules: [src, lib],
    alias: {},
  },
  externals: {},
  output: {
    publicPath,
    filename: 'assets/[name]/[name].js?v=[hash]',
    path: dst,
  },
  entry: {
    common: ['regenerator-runtime/runtime'],
    index: [`${src}/main`],
  },
  module: {
    rules: [
      {
        test: /\.tsx?$/,
        include: [src],
        loader: 'awesome-typescript-loader',
      },
      {
        test: /\.(ico|jpg|jpeg|png|gif|eot|otf|webp|svg|ttf|woff2?)$/,
        include: [src],
        loader: 'url-loader',
        options: {
          name: '[path][name].[ext]',
        },
      },
    ],
  },
  plugins: [
    new HtmlWebpackPlugin({
      inject: true,
      inlineSource: env === 'production' ? '.(js|css)' : false,
      template: 'index.html',
      filename: 'index.html',
    }),
    new webpack.DefinePlugin({
      'process.env.NODE_ENV': JSON.stringify(env || 'development'),
    }),
    new webpack.optimize.CommonsChunkPlugin({
      name: 'common',
      filename: 'assets/[name]/[name].js',
      minChunks: Infinity,
    }),
  ],
}

const production = Object.assign({}, development, {
  devtool: 'source-map',
  devServer: undefined,
})

module.exports = Object.assign({}, { development }, { production })[env]
