// smss/src/style/index.ts
// @大小
export const s = (flex: number | number) => ({ flex })
export const w = (width: string | number) => ({ width })
export const h = (height: string | number) => ({ height })
export const wh = (width: string | number, height: string | number) => ({ width, height: height || width })
// @颜色
export const c = (color: string) => ({ color })
// @背景
export const b = (backgroundColor: string) => ({ backgroundColor })
// @字体
export const f = (fontSize: number) => ({ fontSize })
export const lh = (lineHeight: number) => ({ lineHeight })
// @内边距
export const pd = (padding: number) => ({ padding })
export const pt = (paddingTop: number) => ({ paddingTop })
export const pr = (paddingRight: number) => ({ paddingRight })
export const pb = (paddingBottom: number) => ({ paddingBottom })
export const pl = (paddingLeft: number) => ({ paddingLeft })
export const plr = (paddingLeft: number, paddingRight: number) =>
  ({ paddingLeft, paddingRight: paddingRight || paddingLeft })
export const ptb = (paddingTop: number, paddingBottom: number) =>
  ({ paddingTop, paddingBottom: paddingBottom || paddingTop, })
// @外边距
export const mg = (margin: number) => ({ margin })
export const mt = (marginTop: number) => ({ marginTop })
export const mr = (marginRight: number) => ({ marginRight })
export const mb = (marginBottom: number) => ({ marginBottom })
export const ml = (marginLeft: number) => ({ marginLeft })
export const mlr = (marginLeft: number, marginRight: number) => ({ marginLeft, marginRight: marginRight || marginLeft })
export const mtb = (marginTop: number, marginBottom: number) => ({ marginTop, marginBottom: marginBottom || marginTop })
// @位置
export const ps = (top: number, right: number, bottom: number, left: number) => ({ top, right, bottom, left })
